tests/e2e/unified_test.go

```diff
@@ -170,9 +168,26 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 					return primaryResource, opt
 				}
 
+				runScenario(ctx, t, testPause, fixture, loadFixture)
+			})
+		}
+	})
+
+	// Do a cleanup while we can still handle the error.
+	t.Logf("shutting down manager")
+	cancel()
+}
+
+func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture resourcefixture.ResourceFixture, loadFixture func(project testgcp.GCPProject, uniqueID string) (*unstructured.Unstructured, create.CreateDeleteTestOptions)) {
```

Comment:
But also could we rename this to `runTest`? Scenario makes me think of the step/ scenario based testing but if that's on purpose then never mind 😛

Reply:
Let's have this PR goes in and then we can discuss/improve in follow up changes. 🥂